test/acceptance/registry.js

```diff
@@ -46,6 +51,228 @@ describe('registry', function(){
+        expect(result[0].response.version).to.equal('1.0.0');
+        expect(result[0].response.name).to.equal('hello-world-custom-headers');
+        expect(result[0].response.headers).to.be.deep.equal({'cache-control': 'public max-age=3600', 'test-header': 'Test-Value'});
+      });
+    });
+
+    describe('with a custom configuration with customHeadersToSkipOnWeakVersion defined', function() {
+      before(function(done) {
+        registry.close();
+
+        conf = {          
+          local: true,
+          path: path.resolve('test/fixtures/components'),
+          port: 3030,
+          baseUrl: 'http://localhost:3030/',
+          env: { name: 'local' },
+          verbosity: 0,
+          dependencies: ['underscore'],
+          customHeadersToSkipOnWeakVersion: ['Cache-Control']
+        };
+
+        registry = new oc.Registry(conf);
+        registry.start(done);
+      });
+
+      describe('when strong version is requested 1.0.0', function() {
+        before(function(done) {
+          request({
+            url: 'http://localhost:3030',
+            json: true,
+            method: 'post',
+            body: {
+              components: [{
+                name: 'hello-world-custom-headers',
+                version: '1.0.0'
+              }]
+            }
+          }, next(done));
+        });
+
+        it('should not set HTTP custom headers', function() {
+          expect(headers).to.not.have.property('cache-control');
+          expect(headers).to.not.have.property('test-header');
+        });
+
+        it('should return the component with the custom headers', function() {
+          expect(result[0].response.version).to.equal('1.0.0');
+          expect(result[0].response.name).to.equal('hello-world-custom-headers');
+          expect(result[0].response.headers).to.be.deep.equal({'cache-control': 'public max-age=3600', 'test-header': 'Test-Value'});
+        });
+      });
+
+      describe('when weak version is requested 1.x.x', function() {
+        before(function(done) {
+          request({
+            url: 'http://localhost:3030',
+            json: true,
+            method: 'post',
+            body: {
+              components: [{
+                name: 'hello-world-custom-headers',
+                version: '1.x.x'
+              }]
+            }
+          }, next(done));
+        });
+
+        it('should not set HTTP custom headers', function() {
+          expect(headers).to.not.have.property('cache-control');
+          expect(headers).to.not.have.property('test-header');
+        });
+
+        it('should skip Cache-Control header', function() {
+          expect(result[0].response.version).to.equal('1.0.0');
+          expect(result[0].response.name).to.equal('hello-world-custom-headers');
+          expect(result[0].response.headers).to.be.deep.equal({'cache-control': 'public max-age=3600', 'test-header': 'Test-Value'});
```

Comment:
You are right. It's more on that though. Actually I didn't realise the POST requests go to component*s*.js instead of component.js. Now this is fixed. Also one of the big trouble with the tests for POST requests was that they are following after those for GET and they are working with an already set customHeadersToSkipOnWeakVersion. So I did a slight refactoring of the before/afters there as well.

Apologies for this mess :-)
And can I ask for one more review, please?